### ISSUE TITLE
Support 'Deploy on Netlify'

### ISSUE THREAD

**Issue by damithc** [MarkBind/markbind#479]
We can support 'Deploy on Netlify' ability, as per https://www.netlify.com/docs/deploy-button/

**Comment by damithc** [comment]
> Are we intending for the template to just simply be the contents of `markbind init`?

I guess so, at least for the moment, until we create better templates.

> If so, this should be relatively straightforward, but the template repository should probably reside in [MarkBind](https://github.com/MarkBind) than in any one of our personal accounts.

Yes. Maybe create a proof of concept repo in your own account first?

**Comment by Xenonym** [comment]
I have created a [boilerplate repository here.](https://github.com/Xenonym/markbind-boilerplate)

Here's the Deploy to Netlify button for testing:

[![Deploy to Netlify](https://www.netlify.com/img/deploy/button.svg)](https://app.netlify.com/start/deploy?repository=https://github.com/Xenonym/markbind-boilerplate)

**Comment by damithc** [comment]
> I have created a [boilerplate repository here.](https://github.com/Xenonym/markbind-boilerplate)

Good work @Xenonym. Let's resume this when we have implemented a better starter template.  The current one is a bit sparse. We'll probably need to improve `markbind init` to create a better starter website and then we can reuse it for this purpose. I'll create an issue for that shortly.

**Comment by Crphang** [comment]
@damithc is the current starter sufficient to add this functionality?

**Comment by damithc** [comment]
> @damithc is the current starter sufficient to add this functionality?

At one point I was contemplating a starter such as this https://github.com/MarkBind/init-typical

**Comment by Crphang** [comment]
After #935 has been merge, the current default starter has been updated to: https://crphang.github.io/web/

Let me know if there are some bugs I need to patch on the current starter. I am currently working on supporting initialization with flags like --minimal templates as well.

**Comment by damithc** [comment]
Some suggestions:
![image](https://user-images.githubusercontent.com/1673303/64863437-65d5d980-d667-11e9-95b9-9174c0c10eef.png)

The tooltip should only be for the word `tooltip`.

Point topNav/siteNav links to a single placeholder page (currently, they give 404)

Enable search?

**Comment by Crphang** [comment]
@damithc fixed made with PR https://github.com/MarkBind/markbind/pull/939. 

You can view the changes at https://crphang.github.io/

**Comment by damithc** [comment]
Looks good @Crphang

**Comment by Crphang** [comment]
@damithc Here's one of Deploy on Netlify supported: https://suspicious-mccarthy-2f6d3d.netlify.com/. 

Currently I am hosting it on my own repository: https://github.com/Crphang/webrepo. Would it be good to create one to be hosted within Markbind account? 

Some design decisions on repository: I added the markbind site within `your_site` to have a clear separation of concern between netlify dependencies for deployment (which our user should not care about going forward) and the actual content that the user should be focused on.

**Comment by Crphang** [comment]
Going forward, I see this being very useful for enhancing Markbind templates feature:

![image](https://user-images.githubusercontent.com/4986717/67147804-e6bd6c00-f2ca-11e9-83ae-14d8f0195e2a.png)

This allows our users to quickly deploy and starting working on each template.

**Comment by damithc** [comment]
> Currently I am hosting it on my own repository: https://github.com/Crphang/webrepo. Would it be good to create one to be hosted within Markbind account?

Yes, I think so.

> Some design decisions on repository: I added the markbind site within `your_site` to have a clear separation of concern between netlify dependencies for deployment (which our user should not care about going forward) and the actual content that the user should be focused on.

@yamgent what do you think?


> Going forward, I see this being very useful for enhancing Markbind templates feature:

I think that's a good idea too 👍

**Comment by yamgent** [comment]
> Currently I am hosting it on my own repository: https://github.com/Crphang/webrepo. Would it be good to create one to be hosted within Markbind account?

Sure, if we are going to host multiple templates in the future, might be even better to create a new organization for them, unless Netlify supports putting everything in one repo?

> Some design decisions on repository: I added the markbind site within your_site to have a clear separation of concern between netlify dependencies for deployment (which our user should not care about going forward) and the actual content that the user should be focused on.

Sure, I am fine with that.

**Comment by yamgent** [comment]
@Crphang I have duplicated your repo into https://github.com/MarkBind/init-typical-netlify, you can do any additional changes here.

**Comment by Crphang** [comment]
@yamgent, could we have this https://github.com/Crphang/minimalrepo duplicated as well? I'll then make a PR with one-click deploy on our templates page that is based off the repo hosted in the organization as shown below:

![image](https://user-images.githubusercontent.com/4986717/69635079-8e7a4680-108e-11ea-9d5b-e013f84d20e7.png)

**Comment by yamgent** [comment]
https://github.com/MarkBind/init-minimal-netlify

**Comment by damithc** [comment]
How do we git this to appear for MarkBind?
![image](https://user-images.githubusercontent.com/1673303/69639198-15332180-1097-11ea-84d4-50980c4b0adc.png)

**Comment by yamgent** [comment]
We will need to create another PR to staticgen, with the following changes, to incorporate the "Deploy to Netlify" button:

```diff
diff --git a/content/projects/markbind.md b/content/projects/markbind.md
index df1d82d..2bacde0 100644
--- a/content/projects/markbind.md
+++ b/content/projects/markbind.md
@@ -9,6 +9,7 @@ license:
 templates:
   - Custom
 description: Optimized for generating technical documentation and educational websites from Markdown text.
+startertemplaterepo: MarkBind/init-typical-netlify
 ---

 MarkBind is a website generator that can generate a website from markdown documents. While there are other markdown-to-html website generators around, MarkBind can generate more dynamic websites, as opposed to one-size-fits-all static content. MarkBind is particularly suitable to create content-heavy websites e.g., eLearning websites, online instruction manuals, project documentation etc. that aim for self-directed consumption.
```

**Comment by damithc** [comment]
> We will need to create another PR to staticgen, with the following changes, to incorporate the "Deploy to Netlify" button

@Crphang can?

**Comment by Crphang** [comment]
Create PR on their side:

Demo: https://deploy-preview-654--staticgen.netlify.com/
